### PR TITLE
Fixes #486

### LINKED ISSUE #486: [IFC Tunnel] Add IfcArchElement and IfcArchElementType
These seem to have been forgotten during the previous IFC Tunnel extension session.
ENTITY IfcArchElement
SUBTYPE OF IfcBuiltElement
Definition: A unitary curved structure.

ENTITY IfcArchElementType
SUBTYPE of  IfcBuiltElementType
Definition: The element type IfcArchElementType defines commonly shared information for occurrences of arch elements.

And the usual CorrectPredefinedType/CorrectTypeAssigned (Only USERDEFINED and NOTDEFINED as of yet)

### PR DESCRIPTION
**Pull request by larswik**
Fixes #486. Also saw that IfcFillElementTypeEnum lacked description so I added that as well here.

### CODE REVIEW

File: IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedBldgElements/Entities/IfcArchElementType/Documentation.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+The element type _IfcArchElementType_ defines commonly shared information for occurrences of arch elements. Arch elements are building elements which represents unitary curved structures.The set of shared information may include:
```
Comment (pjanck) [inline IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedBldgElements/Entities/IfcArchElementType/Documentation.md]:
```suggestion
The element type _IfcArchElementType_ defines commonly shared information for occurrences of arch elements. Arch elements are building elements which represents unitary curved structures. The set of shared information may include:
```

File: IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedBldgElements/Entities/IfcArchElementType/Documentation.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+
+* common properties within shared property sets
+* common material information
+* common profile definitions
+* common shape representations
+
+It is used to define an arch element specification that is common to all occurrences of that arch element type. Arch element types may be exchanged without being already assigned to occurrences.
+
+Occurrences of the _IfcFillElementType_ within building models are represented by instances of _IfcFillElement_.
```
Comment (pjanck) [inline IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedBldgElements/Entities/IfcArchElementType/Documentation.md]: *Fill*?

**Review by pjanck — CHANGES_REQUESTED** [review]

File: IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedBldgElements/Types/IfcArchElementTypeEnum/Documentation.md [hunk 3]
```diff
@@ -0,0 +1 @@
+This enumeration defines the different types of elements an _IfcArchElement_ or _IfcArchElementType_ object can fulfill.
```
Comment (SergejMuhic) [inline IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedBldgElements/Types/IfcArchElementTypeEnum/Documentation.md]: Feel free to reject. Just thought it sounds better and fits better with the relationship.
```suggestion
This enumeration defines the different types of elements an _IfcArchElement_ or _IfcArchElementType_ object can be defined as.
```

File: IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedBldgElements/Types/IfcFillElementTypeEnum/Documentation.md [hunk 4]
```diff
@@ -0,0 +1 @@
+This enumeration defines the different types of elements an _IfcFillElement_ or _IfcFillElementType_ object can fulfill.
```
Comment (SergejMuhic) [inline IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedBldgElements/Types/IfcFillElementTypeEnum/Documentation.md]:
```suggestion
This enumeration defines the different types of elements an _IfcFillElement_ or _IfcFillElementType_ object can be defined as.
```

File: IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedBldgElements/Entities/IfcArchElementType/Documentation.md (hunk 2 above)
Comment (SergejMuhic) [reply IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedBldgElements/Entities/IfcArchElementType/Documentation.md]:
```suggestion
Occurrences of the _IfcArchElementType_ within building models are represented by instances of _IfcArchElement_.
```

File: IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedBldgElements/Entities/IfcArchElementType/DocEntity.xml [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+<DocEntity xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="IfcArchElementType" UniqueId="26ea73e4-79d9-4475-af8f-61077ac617c7" EntityFlags="32">
+	<Attributes>
+		<DocAttribute Name="PredefinedType" UniqueId="0ad38127-28ed-41e7-964e-2150e60c4bfa" DefinedType="IfcArchElementTypeEnum">
+			<Documentation>Specifies the type for which the value is selected from a predefined type enumeration.</Documentation>
+		</DocAttribute>
+	</Attributes>
+	<WhereRules>
+		<DocWhereRule Name="CorrectPredefinedType" UniqueId="9e780a13-680e-4fce-939a-948f87163eee">
+			<Documentation>The inherited attribute _ElementType_ shall be provided, if the _PredefinedType_ is set to USERDEFINED.</Documentation>
+			<Expression>(PredefinedType &lt;&gt; IfcArchElementTypeEnum.USERDEFINED) OR
```
Comment (SergejMuhic) [inline IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedBldgElements/Entities/IfcArchElementType/DocEntity.xml:11]: I think I should turn off this html encoding. Really looks bad. 😄

File: IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedBldgElements/Entities/IfcArchElement/Documentation.md [hunk 6]
```diff
@@ -0,0 +1 @@
+A unitary curved structure.
```
Comment (SergejMuhic) [inline IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedBldgElements/Entities/IfcArchElement/Documentation.md]: I like Meriam Webster: "a typically curved structural member spanning an opening and serving as a support"

https://www.merriam-webster.com/dictionary/arch

File: IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedBldgElements/Entities/IfcArchElement/DocEntity.xml [hunk 7]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocEntity xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="IfcArchElement" UniqueId="497395c2-219d-4d42-9eaf-1a8c3f3967cd" EntityFlags="32">
+	<Attributes>
+		<DocAttribute Name="PredefinedType" UniqueId="140250ca-4d21-4dca-9f29-7b99e1c6857b" DefinedType="IfcArchElementTypeEnum" AttributeFlags="1">
+			<Documentation>Predefined generic type for an arch element that is specified in an enumeration. There may be a property set given specifically for the predefined types.
```
Comment (SergejMuhic) [inline IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedBldgElements/Entities/IfcArchElement/DocEntity.xml]: Did we not have a more generic common description that we used? This, seems to me, is not it. 🤔

**Review by SergejMuhic — CHANGES_REQUESTED** [review]

File: IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedBldgElements/Entities/IfcArchElement/Documentation.md (hunk 6 above)
Comment (larswik) [reply IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedBldgElements/Entities/IfcArchElement/Documentation.md]: I agree but I just used the definition from IfcElementAssembly/ARCH and added "unitary". Should we change there as well then?

File: IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedBldgElements/Entities/IfcArchElement/DocEntity.xml (hunk 7 above)
Comment (larswik) [reply IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedBldgElements/Entities/IfcArchElement/DocEntity.xml]: There are a few...A generic one I found is: "Specifies the type for which the value is selected from a predefined type enumeration. This type may associate additional specific property sets".

File: IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedBldgElements/Entities/IfcArchElement/Documentation.md (hunk 6 above)
Comment (SergejMuhic) [reply IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedBldgElements/Entities/IfcArchElement/Documentation.md]: I would not dare. That is bSI business.

File: IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedBldgElements/Entities/IfcArchElement/DocEntity.xml (hunk 7 above)
Comment (SergejMuhic) [reply IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedBldgElements/Entities/IfcArchElement/DocEntity.xml]: This wording "Predefined generic type" bothers me a bit. What is generic about the predefined type? 🤔

**Review by pjanck — APPROVED** [review]
lftm

File: IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedBldgElements/Entities/IfcArchElement/DocEntity.xml (hunk 7 above)
Comment (larswik) [reply IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedBldgElements/Entities/IfcArchElement/DocEntity.xml]: > This wording "Predefined generic type" bothers me a bit. What is generic about the predefined type? 🤔

Please check now.

File: IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedBldgElements/Entities/IfcArchElement/Documentation.md (hunk 6 above)
Comment (larswik) [reply IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedBldgElements/Entities/IfcArchElement/Documentation.md]: > I like Meriam Webster: "a typically curved structural member spanning an opening and serving as a support"
> 
> https://www.merriam-webster.com/dictionary/arch

Please check now.

File: IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedBldgElements/Entities/IfcArchElementType/Documentation.md [hunk 8]
```diff
@@ -0,0 +1,10 @@
+The element type _IfcArchElementType_ defines commonly shared information for occurrences of arch elements. Arch elements are building elements which represents unitary curved structures. The set of shared information may include:
```
Comment (anborr) [inline IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedBldgElements/Entities/IfcArchElementType/Documentation.md]:
```suggestion
The element type _IfcArchElementType_ defines commonly shared information for occurrences of arch elements. Arch elements are building elements which represent unitary curved structures. The set of shared information may include:
```

**Review by anborr — APPROVED** [review]

File: IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedBldgElements/Entities/IfcArchElement/Documentation.md [hunk 9]
```diff
@@ -0,0 +1 @@
+A typically curved structural member spanning an opening and serving as a support
```
Comment (SergejMuhic) [inline IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedBldgElements/Entities/IfcArchElement/Documentation.md]:
```suggestion
A typically curved structural member spanning an opening and serving as a support.
```

**Review by SergejMuhic — APPROVED** [review]

**Comment by SergejMuhic** [comment]
WIll merge after schema generation with commit hash has been implemented.

**Pull request merged**